client/dashboard/components/metadata-list/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { __experimentalHStack as HStack, __experimentalText as Text } from '@wordpress/components';
+import './style.scss';
+
+interface MetadataListProps {
+	children?: React.ReactNode;
+}
+
+type MetadataItemProps =
+	| { title: React.ReactNode; children?: React.ReactNode }
```

Comment:
I think the children is always required so maybe we can remove this 🤔

Reply:
This is mostly for developer ergonomics to communicate that you need to pass something. It could be removed, I guess, but since types are removed at compile time. I think this is a good reminder.